src/go/languageservice/workspace.go

```diff
@@ -30,334 +30,288 @@ type WorkspaceID int
 }
 
 type completionsFindingVisitor struct {
-	SourcePos         token.Pos
-	PositionSensitive bool
-	Completions       *[]string
+    SourcePos         token.Pos
+    PositionSensitive bool
+    Completions       *[]string
 }
 
 func (v completionsFindingVisitor) Visit(node ast.Node) ast.Visitor {
-	switch nodeCasted := node.(type) {
-	case *ast.FuncDecl:
-		*v.Completions = append(*v.Completions, nodeCasted.Name.Name)
-		return nil // don't go deeper
-	case *ast.ValueSpec:
-		if v.PositionSensitive && node.Pos() >= v.SourcePos {
-			return nil
-		}
-
-		for _, name := range nodeCasted.Names {
-			if name != nil {
-				*v.Completions = append(*v.Completions, name.Name)
-			}
-		}
-		return nil
-	case *ast.TypeSpec:
-		if nodeCasted.Name != nil {
-			*v.Completions = append(*v.Completions, nodeCasted.Name.Name)
-			return nil
-		}
-	case *ast.ImportSpec:
-		if nodeCasted.Name != nil {
-			*v.Completions = append(*v.Completions, nodeCasted.Name.Name)
-			return nil
-		} else if nodeCasted.Path != nil {
-			*v.Completions = append(*v.Completions, nodeCasted.Path.Value[1:len(nodeCasted.Path.Value)-1])
-			return nil
-		}
-	case *ast.AssignStmt:
-		if v.PositionSensitive && node.Pos() >= v.SourcePos {
-			return nil
-		}
-
-		identifier := nodeCasted.Lhs[0].(*ast.Ident)
-		if identifier != nil {
-			*v.Completions = append(*v.Completions, identifier.Name)
-		}
-		return nil
-	}
-	return v
+    switch nodeCasted := node.(type) {
+    case *ast.FuncDecl:
+        *v.Completions = append(*v.Completions, nodeCasted.Name.Name)
+        return nil // don't go deeper
+    case *ast.ValueSpec:
+        if v.PositionSensitive && node.Pos() >= v.SourcePos {
+            return nil
+        }
+
+        for _, name := range nodeCasted.Names {
+            if name != nil {
+                *v.Completions = append(*v.Completions, name.Name)
+            }
+        }
+        return nil
+    case *ast.TypeSpec:
+        if nodeCasted.Name != nil {
+            *v.Completions = append(*v.Completions, nodeCasted.Name.Name)
+            return nil
+        }
+    case *ast.ImportSpec:
+        if nodeCasted.Name != nil {
+            *v.Completions = append(*v.Completions, nodeCasted.Name.Name)
+            return nil
+        } else if nodeCasted.Path != nil {
+            *v.Completions = append(*v.Completions, nodeCasted.Path.Value[1:len(nodeCasted.Path.Value)-1])
+            return nil
+        }
+    case *ast.AssignStmt:
+        if v.PositionSensitive && node.Pos() >= v.SourcePos {
+            return nil
+        }
+
+        identifier := nodeCasted.Lhs[0].(*ast.Ident)
+        if identifier != nil {
+            *v.Completions = append(*v.Completions, identifier.Name)
+        }
+        return nil
+    }
+    return v
 }
 
 type TokenType int
 
 const (
-	KEYWORD      TokenType = 1
-	IDENTIFIER   TokenType = 2
-	STRING       TokenType = 3
-	RESOLVEDTYPE TokenType = 4
-	LITERAL      TokenType = 5
-	COMMENT      TokenType = 6
+    KEYWORD = iota
```

Comment:
Are these guaranteed ascending?

Note that I hard coded integer values here to ensure they match the C# side if reordered

Reply:
yes they are

Reply:
Ok, it's definitely worth a comment at least to ensure they don't get messed up.

I'd actually feel a bit more comfortable leaving the int values in place because this is a binary compatibility issue and not just a source code one.

Reply:
Yea, I'd agree with that, will change back. Perhaps move their declarations to the token_service with a comment as well?